include/h2/gpu/cuda/memory_utils.hpp

```diff
@@ -13,18 +13,16 @@
  *  Thin wrappers around cudaMem{cpy,set} functions. These are here so
  *  they can be inlined if possible.
  */
+#if H2_HAS_CUDA
+#include "hydrogen/PoolAllocator.hpp"
+#endif
```

Comment:
```suggestion
#include <hydrogen/PoolAllocator.hpp>
```
This file shouldn't be included unless `H2_HAS_CUDA`.